File: Dataset/Leetcode/valid/101/737.py
 class Solution:
    def XXX(self, root: TreeNode) -> bool:
        def isSym(left,right):
            if left and right:
                return left.val == right.val and isSym(left.right, right.left) and isSym(left.left, right.right)
            else:
                return left == right
        return isSym(root, root)

File: Dataset/Leetcode/train/112/737.cpp
 class Solution {
public:
    bool XXX(TreeNode* root, int sum) {
        if(root==NULL)return false;
        
        return dfs(root,0,sum);
    }

   bool dfs(TreeNode* root,int count,int sum){
       if(root==NULL)return false;
       count+=root->val;
        if(root->left==NULL&&root->right==NULL){
            if(count==sum){
                return true;
            }
        }
        
        return dfs(root->left,count,sum)||dfs(root->right,count,sum);  
    }
};

